src/overlay.cpp

```diff
@@ -694,24 +694,40 @@ void init_system_info(){
 
       wineProcess = get_exe_path();
       trim(wineProcess);
-      string searchPreload("preloader");
-      if (wineProcess.find(searchPreload) != std::string::npos) {
-         stringstream findPath;
-         findPath <<"echo " << wineProcess << " | sed 's/^/\"/' " <<" | sed 's%/[^/]*$%/wine\"%' ";
-         wineProcess = exec(findPath.str());
-         trim(wineProcess);
-         bool env_exists = false;
-         if (getenv("WINELOADERNOEXEC")) {
-            static char removenoexec[] = "WINELOADERNOEXEC";
-            putenv(removenoexec);
-            env_exists = true;
+      auto n = wineProcess.find_last_of('/');
+      string preloader = wineProcess.substr(n + 1);
+      string searchProton("/dist/bin/wine");
+      if (preloader == "wine-preloader" || preloader == "wine64-preloader") {
+         // Check if using Proton
+         if (wineProcess.find(searchProton) != std::string::npos) {
+            stringstream protonPath;
+            protonPath << "echo " << wineProcess <<" | rev | cut -d '/' -f3- | rev | sed 's/.*/\"&\"/'";
+            string protonVersion;
+            protonVersion = exec(protonPath.str());
+            trim(protonVersion);
+            stringstream command;
+            command <<"< " <<protonVersion + "/version" <<" cut -d' ' -f2";
```

Comment:
Thinking something like this to make it prettier and conform more to how it's displayed in Steam
```
command <<"< " <<protonVersion + "/version" <<" cut -d' ' -f2 | sed -e 's|proton-||' ";
wineVersion = exec(command.str());
wineVersion = "Proton " + wineVersion;
```

Reply:
Yes that is much nicer.. also was thinking what if we replace WINE with PROTON and then  display the proton version underneath?
`` ImGui::TextColored(wine_color, "%s", "PROTON");``

Reply:
I'm not sure the WINE/PROTON text is required, the version should always contain proton or wine in it already